File: go/000/ListNode.go
package main

import "fmt"

type ListNode struct {
    Val int
    Next *ListNode
}

func getListNode(ints ...int) *ListNode {
    var re *ListNode = &ListNode{ints[0], nil}
    prev := re
    for i, n := 1, len(ints); i < n; i++ {
        prev.Next = &ListNode{ints[i], nil}
        prev = prev.Next
    }
    return re
}

func printListNode(nodes ...*ListNode)  {
    for _, node := range nodes {
        for node != nil {
            fmt.Print(node.Val)
            fmt.Print("\t")
            node = node.Next;
        }
        fmt.Print("\n")
    }
}
//go build ListNode.go && ListNode && go clean
func main()  {
    printListNode(getListNode(1, 5, 6, 2), getListNode(7, 8, 6, 2))
}

File: swift/000/ListNode.swift

public class ListNode {
    public var val: Int
    public var next: ListNode?
    public init(_ val: Int) {
        self.val = val
        self.next = nil
    }

    class func getListNode(ints: Int...) -> ListNode {
        let re = ListNode(ints[0])
        var prev: ListNode = re
        for i in 1..<ints.count {
            prev.next = ListNode(ints[i])
            prev = prev.next!
        }
        return re
    }

    class func printListNode(nodes: ListNode...) {
        var bytes: [Character] = []
        for node in nodes {
            var nl: ListNode? = node
            while nl != nil {
                bytes.append(Character(UnicodeScalar((nl!.val) + 48)))
                bytes.append("\t")
                nl = nl!.next
            }
            bytes.append("\n")
        }
        print(String(bytes))
    }
}

ListNode.printListNode(ListNode.getListNode(1, 5, 6, 2), ListNode.getListNode(7, 8, 6, 2));
